pyscriptjs/src/runtime.ts

```diff
@@ -0,0 +1,101 @@
+import type { loadPyodide } from 'pyodide';
+
+// The current release doesn't export `PyodideInterface` type
+export type PyodideInterface = Awaited<ReturnType<typeof loadPyodide>>;
+export type RuntimeInterpreter = PyodideInterface | null;
+
+/*
+Runtime class is a super class that all different runtimes must respect
+and adhere to.
+
+Currently, the only runtime available is Pyodide as indicated by the
+`RuntimeInterpreter` type above. This serves as a Union of types of
+different runtimes/interpreters which will be added in near future.
+
+The class has abstract methods available which each runtime is supposed
+to implement. The current design is based on the Pyodide runtime (since
+that is the only one available), but is subject to change.
+
+Methods available handle loading of the interpreter, initialization,
+running code, getting the globals symbol table, loading and installation
+of packages, loading from files etc.
+
+For an example implementation, refer to the `PyodideRuntime` class
+in `pyodide.ts`
+*/
+export abstract class Runtime extends Object {
```

Comment:
Right now it's an interface... I think there's some logic that we can provide to all the runtimes that subclass from it though

Reply:
yes, we can move all `initializers`, `postInitializers`, `scriptsQueue`, etc. here, done that